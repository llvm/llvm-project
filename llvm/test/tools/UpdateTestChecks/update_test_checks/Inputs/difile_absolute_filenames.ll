; RUN: opt < %s -S | FileCheck %s

source_filename = "/llvm-project/foo.c"

define dso_local void @f() !dbg !10 {
entry:
  ret void, !dbg !14
}

!llvm.dbg.cu = !{!0}
!llvm.module.flags = !{!2, !3, !4, !5, !6, !7, !8}
!llvm.ident = !{!9}

!0 = distinct !DICompileUnit(language: DW_LANG_C11, file: !1, producer: "clang", isOptimized: false, runtimeVersion: 0, emissionKind: FullDebug, splitDebugInlining: false, nameTableKind: None)
!1 = !DIFile(filename: "/llvm-project/foo.c", directory: "/llvm-project")
!2 = !{i32 7, !"Dwarf Version", i32 5}
!3 = !{i32 2, !"Debug Info Version", i32 3}
!4 = !{i32 1, !"wchar_size", i32 4}
!5 = !{i32 8, !"PIC Level", i32 2}
!6 = !{i32 7, !"PIE Level", i32 2}
!7 = !{i32 7, !"uwtable", i32 2}
!8 = !{i32 7, !"frame-pointer", i32 2}
!9 = !{!"clang"}
!10 = distinct !DISubprogram(name: "f", scope: !11, file: !11, line: 1, type: !12, scopeLine: 1, spFlags: DISPFlagDefinition, unit: !0)
!11 = !DIFile(filename: "foo.c", directory: "/llvm-project")
!12 = !DISubroutineType(types: !13)
!13 = !{null}
!14 = !DILocation(line: 1, column: 11, scope: !10)
