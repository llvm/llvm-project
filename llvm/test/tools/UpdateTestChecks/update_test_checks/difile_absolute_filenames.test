## Test that update_test_checks.py is not sensitive to absolute paths in
## DIFile's filename field.
# RUN: cp -f %S/Inputs/difile_absolute_filenames.ll %t.ll && %update_test_checks %t.ll --check-globals
# RUN: diff -u %t.ll %S/Inputs/difile_absolute_filenames.ll.expected
## Check that running the script again does not change the result:
# RUN: %update_test_checks %t.ll --check-globals
# RUN: diff -u %t.ll %S/Inputs/difile_absolute_filenames.ll.expected
