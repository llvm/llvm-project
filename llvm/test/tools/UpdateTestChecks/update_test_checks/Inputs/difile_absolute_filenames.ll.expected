; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --check-globals
; RUN: opt < %s -S | FileCheck %s

source_filename = "/llvm-project/foo.c"

define dso_local void @f() !dbg !10 {
; CHECK-LABEL: @f(
; CHECK-NEXT:  entry:
; CHECK-NEXT:    ret void, !dbg [[DBG14:![0-9]+]]
;
entry:
  ret void, !dbg !14
}

!llvm.dbg.cu = !{!0}
!llvm.module.flags = !{!2, !3, !4, !5, !6, !7, !8}
!llvm.ident = !{!9}

!0 = distinct !DICompileUnit(language: DW_LANG_C11, file: !1, producer: "clang", isOptimized: false, runtimeVersion: 0, emissionKind: FullDebug, splitDebugInlining: false, nameTableKind: None)
!1 = !DIFile(filename: "/llvm-project/foo.c", directory: "/llvm-project")
!2 = !{i32 7, !"Dwarf Version", i32 5}
!3 = !{i32 2, !"Debug Info Version", i32 3}
!4 = !{i32 1, !"wchar_size", i32 4}
!5 = !{i32 8, !"PIC Level", i32 2}
!6 = !{i32 7, !"PIE Level", i32 2}
!7 = !{i32 7, !"uwtable", i32 2}
!8 = !{i32 7, !"frame-pointer", i32 2}
!9 = !{!"clang"}
!10 = distinct !DISubprogram(name: "f", scope: !11, file: !11, line: 1, type: !12, scopeLine: 1, spFlags: DISPFlagDefinition, unit: !0)
!11 = !DIFile(filename: "foo.c", directory: "/llvm-project")
!12 = !DISubroutineType(types: !13)
!13 = !{null}
!14 = !DILocation(line: 1, column: 11, scope: !10)
;.
; CHECK: [[META0:![0-9]+]] = distinct !DICompileUnit(language: DW_LANG_C11, file: [[META1:![0-9]+]], producer: "clang", isOptimized: false, runtimeVersion: 0, emissionKind: FullDebug, splitDebugInlining: false, nameTableKind: None)
; CHECK: [[META1]] = !DIFile(filename: "{{.*}}foo.c", directory: {{.*}})
; CHECK: [[META2:![0-9]+]] = !{i32 7, !"Dwarf Version", i32 5}
; CHECK: [[META3:![0-9]+]] = !{i32 2, !"Debug Info Version", i32 3}
; CHECK: [[META4:![0-9]+]] = !{i32 1, !"wchar_size", i32 4}
; CHECK: [[META5:![0-9]+]] = !{i32 8, !"PIC Level", i32 2}
; CHECK: [[META6:![0-9]+]] = !{i32 7, !"PIE Level", i32 2}
; CHECK: [[META7:![0-9]+]] = !{i32 7, !"uwtable", i32 2}
; CHECK: [[META8:![0-9]+]] = !{i32 7, !"frame-pointer", i32 2}
; CHECK: [[META9:![0-9]+]] = !{!"clang"}
; CHECK: [[META10:![0-9]+]] = distinct !DISubprogram(name: "f", scope: [[META11:![0-9]+]], file: [[META11]], line: 1, type: [[META12:![0-9]+]], scopeLine: 1, spFlags: DISPFlagDefinition, unit: [[META0]])
; CHECK: [[META11]] = !DIFile(filename: "{{.*}}foo.c", directory: {{.*}})
; CHECK: [[META12]] = !DISubroutineType(types: [[META13:![0-9]+]])
; CHECK: [[META13]] = !{null}
; CHECK: [[DBG14]] = !DILocation(line: 1, column: 11, scope: [[META10]])
;.
